IPA-End-to-End.md

```diff
@@ -1924,6 +1927,46 @@ An example of an oblivious algorithm is the bitonic sorting algorithm [[1]](http
 The last-touch attribution algorithm described above is oblivious. The algorithm can be parallelized into a large number of small computations that only compare two rows at a time. The pairs of rows which will be compared are always the same, regardless of the values of the input. As such, the _helper party_ learns nothing about the input, and can easily parallelize all of these comparisons.
 
 
+## Webviews and Embedded Browsers
+
+On some operating systems, browser components can be integrated into applications in a number of ways.
+Two models are considered here:
+
+1. Where the embedded browser component permits the application to access (or modify) content.
+
+2. Where the operation of the browser component is not accessible to the application.
+
+### Applications that can access web content
+
+If the embedded browser makes content accessible to the application, then conceptually the application needs to act as an independent browser.
+The application should have distinct storage from the regular browser so that the application is unable to access cookies or other data that is shared by users and sites.
```

Comment:
Recommend differentiating between storage of *match keys* and caches of encrypted-secret-shared match keys.